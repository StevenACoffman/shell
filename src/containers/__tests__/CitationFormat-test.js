import React from "react";
import CitationFormat from "../CitationFormat.jsx";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import {initializeState} from "./outline-test.js";

const mockStore = configureStore([]);
const store = mockStore(initializeState());

it("renders correctly", () => {
    const tree = renderer.create(
        <Provider store={store}>
            <CitationFormat/>
        </Provider>
  ).toJSON();
    expect(tree).toMatchSnapshot();
});
