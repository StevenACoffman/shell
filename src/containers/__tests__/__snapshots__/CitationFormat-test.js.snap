exports[`test renders correctly 1`] = `
<div>
  <div
    className="button-flex-row">
    <button
      className="button button-jstor mrl"
      data-capn-logger="{\"uuid\":\"\",\"engagement_kpi_v1\":true,\"myjstor_userid\":\"mockuser\",\"mylists_list_id\":1,\"event_type\":\"download_outline\"}"
      id="download-outline"
      onClick={[Function]}
      type="button">
      Download
    </button>
    <div>
      <div
        className="Dropdown-root">
        <div
          className="Dropdown-control "
          onMouseDown={[Function]}
          onTouchEnd={[Function]}>
          <div
            className="Dropdown-placeholder">
            Select...
          </div>
          <span
            className="Dropdown-arrow" />
        </div>
      </div>
      <ul
        className="list-jstor hide" />
    </div>
    <div
      aria-live="polite"
      className="mhl mtl mbs blue-text"
      data-capn-logger="{\"uuid\":\"\",\"engagement_kpi_v1\":true,\"myjstor_userid\":\"mockuser\",\"mylists_list_id\":1,\"event_type\":\"save_outline\"}"
      id="save-outline">
      All changes saved.
    </div>
  </div>
  <hr />
  <div
    className="row">
    <div
      className="columns">
      <div
        id="thesis-component">
        <label
          className="bold"
          htmlFor="thesis-body">
          Thesis Statement
        </label>
        <textarea
          id="thesis-body"
          onChange={[Function]}
          placeholder="Placeholder..."
          rows="5"
          type="text"
          value="" />
      </div>
      <div>
        <div
          className="row">
          <hr
            className="mhm" />
          <div
            className="small-9 columns">
            <div>
              <label
                className="bold"
                htmlFor="name_0">
                Section Name
              </label>
              <input
                id="name_0"
                onChange={[Function]}
                type="text"
                value="" />
            </div>
            <div
              className="row">
              <div
                className="small-10 small-offset-2 columns">
                <div>
                  <label
                    className="bold"
                    htmlFor="notes_0">
                    Section Notes
                  </label>
                  <textarea
                    id="notes_0"
                    name="notes"
                    onChange={[Function]}
                    rows="5"
                    value="" />
                </div>
                <div>
                  <div>
                    <span
                      className="subheader visually-hidden">
                      Citations
                    </span>
                    <ul
                      className="list-jstor mbn"
                      id="citations_0" />
                  </div>
                </div>
                <div>
                  <button
                    className="btn-link mhn phn"
                    data-reveal-id="sectionModal_0"
                    id="add_citations_section_0"
                    onClick={[Function]}>
                    + Add Citation From List
                  </button>
                  <noscript />
                </div>
              </div>
            </div>
          </div>
          <div
            className="small-3 columns">
            <div
              className="row mtl">
              <div
                className="small-4 columns outline__tooltip">
                <button
                  className="outline-section__button outline-section__button--move-up secondary disabled"
                  disabled={true}
                  id="move_section_up_button_0"
                  onClick={[Function]}
                  type="button" />
                <span
                  className="outline__tooltiptext">
                  <div
                    className="outline__tooltip-uparrow" />
                  <div
                    className="mtm">
                    Move section up
                  </div>
                </span>
              </div>
              <div
                className="small-4 columns outline__tooltip">
                <button
                  className="outline-section__button outline-section__button--move-down secondary disabled"
                  disabled={true}
                  id="move_section_down_button_0"
                  onClick={[Function]}
                  type="button" />
                <span
                  className="outline__tooltiptext">
                  <div
                    className="outline__tooltip-uparrow" />
                  <div
                    className="mtm">
                    Move section down
                  </div>
                </span>
              </div>
              <div
                className="small-4 columns outline__tooltip">
                <div>
                  <button
                    className="outline-section__button outline-section__button--delete"
                    id="delete_section_0"
                    onClick={[Function]} />
                  <noscript />
                </div>
                <span
                  className="outline__tooltiptext">
                  <div
                    className="outline__tooltip-uparrow" />
                  <div
                    className="mtm">
                    Delete section
                  </div>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <hr />
      <button
        id="add-section"
        onClick={[Function]}>
        + Add Section
      </button>
    </div>
  </div>
</div>
`;

exports[`test renders correctly 2`] = `
<div>
  <div
    className="Dropdown-root">
    <div
      className="Dropdown-control "
      onMouseDown={[Function]}
      onTouchEnd={[Function]}>
      <div
        className="Dropdown-placeholder">
        Select...
      </div>
      <span
        className="Dropdown-arrow" />
    </div>
  </div>
  <ul
    className="list-jstor hide" />
</div>
`;
